mqtt/mqtt-broker/Cargo.toml

```diff
@@ -38,9 +36,10 @@ atty = "0.2"
 bytes = "0.5"
 criterion = "0.3"
 fail = { version = "0.3", features = ["failpoints"] }
+itertools = "0.9"
 matches = "0.1"
 proptest = "0.9"
-rand = "0.3"
+rand = "0.7"
```

Comment:
We should probably put these dependencies behind the feature flag as well.

Reply:
Not blocking an approval on this. But, something we might want to consider to keep it clean.

Reply:
this is a good idea, but it seems like it doesn't work with stable rust now https://github.com/rust-lang/cargo/issues/5954

Reply:
Fixed with `benches` feature